### ISSUE TITLE
Add Java8-Optional functionality to retrofit2-clients

### ISSUE THREAD

**Issue by uschi2000** [palantir/conjure-java-runtime#326]

**Comment by uschi2000** [comment]
As it stands, retrofit does this with Optional parameters:
```
Expected :"/api/getGuavaOptionalString/pathString/?queryString=queryString"
Actual   :"/api/getGuavaOptionalString/Optional.of(pathString)/?queryString=Optional.of(queryString)"
```

**Comment by uschi2000** [comment]
Arguably not what we want

**Comment by uschi2000** [comment]
Will make a converter.

**Comment by uschi2000** [comment]
fixed by #331
